src/test/scala/dibl/TileLinksSpec.scala

```diff
@@ -0,0 +1,62 @@
+package dibl
+
+import org.scalatest.{ FlatSpec, Matchers }
+
+class TileLinksSpec extends FlatSpec with Matchers {
+  // Config.linksOfCenterTile calls Diagram.tileLinks
+  // confusion about the sortBy in the tested method makes lower level tests hard to understand
+
+  "Diagram.tileLinks" should "work for a thread diagram" in {
+    val config = TilesConfig("patchWidth=6&patchHeight=5&tile=5-&tileStitch=ct&&shiftColsSW=-1&shiftRowsSW=1&shiftColsSE=1&shiftRowsSE=1")
+    val diagram = ThreadDiagram.create(NewPairDiagram.create(config))
+    config.linksOfCenterTile(diagram, 2)
+      .map { case (core, clockWise: Array[NodeProps]) =>
+        core.id -> clockWise.map(_.id).mkString(",")
+      } shouldBe Array(
+      "a12" -> "a10,a11,a10,a11",
+      "a11" -> "a12,a10,a12,a10",
+      "a10" -> "a12,a11,a12,a11",
```

Comment:
@veronika you could also just convince yourself with the expected values of the tests in this class.
![](https://raw.githubusercontent.com/d-bl/GroundForge/9fafb4108962fa8100611e3ae073188992eb7117/src/test/scala/dibl/LinkedNodeSpec.png)